src/main/groovy/com/codebrig/arthur/observe/structure/literal/JavaLiteral.groovy

```diff
@@ -9,21 +9,26 @@ import com.codebrig.arthur.observe.structure.StructureLiteral
  * @version 0.4
  * @since 0.2
  * @author <a href="mailto:brandon.fergerson@codebrig.com">Brandon Fergerson</a>
+ * @author <a href="mailto:valpecaoco@gmail.com">Val Pecaoco</a>
  */
 class JavaLiteral extends StructureLiteral {
 
     @Override
     String getNodeLiteralAttribute(SourceNode node) {
         switch (Objects.requireNonNull(node).internalType) {
             case "NumberLiteral":
-                if (node.token.contains(".")
-                        || node.token.toUpperCase().contains("P")
-                        || node.token.toUpperCase().contains("E")
-                        || node.token.toUpperCase().endsWith("D")
-                        || node.token.toUpperCase().endsWith("F")) {
+                if (node.token.contains(".") ||
+                        (node.token.isDouble() &&
+                                (node.token.toUpperCase().contains("P")
```

Comment:
If `node.token.isDouble()` returns true when it's a double, is the `node.token.toUpperCase().contains()` part even necessary?

Reply:
Babelfish actually resolves everything to a double, so something like 100, semantically an integer, returns 100.0 and the `isDouble` check returns true. Above check avoids these integer literals.